### DISCUSSION TITLE
Tif to Jpeg fails

### DISCUSSION THREAD

**Discussion by eduh** [ImageMagick/ImageMagick#6919] [Digital Image Processing]
I am using a small php script to convert tif's to jpg. Already done 4 millions of images with success but now I encounter problems with some tifs with extraordinary sizes.  
The test image is on https://surfdrive.surf.nl/files/index.php/s/JhA5IhPVz8GJAfx  (approx 1.8 G)
The php script (version 8) on Debian  is straight forward, something like this
```
$testtif="17091284535385353799527929296507552950.tif";
$image = new Imagick($testtif);
$image->resizeImage($sizes[0]*(2/3), $sizes[1]*(2/3), Imagick::FILTER_LANCZOS,1); 
$image->setImageCompression(Imagick::COMPRESSION_JPEG);
$image->setImageCompressionQuality(50);
$image->writeImage('17091284535385353799527929296507552950.jpg');
```
The resize fails. I tried several other strategies eg without resize but then the write fails (with or without compression)
What I noticed is that getimagewidth+height are returning wrong dimensions. They should be 16751x19001 but imagemagick is doing something like 129x11437
When I convert with a testtool like irfanview there is no problem in converting so the tif looks alright to me.

**Comment by snibgo** [comment]
1.8 GB is too large for me to download.

What are the values of `$sizes[0]` and `$sizes[1]`?

> The resize fails.

How do you know it fails? What exception is raised?

**Reply by eduh** [reply]
see reply below

**Comment by eduh** [comment]
The resize raises this Deprecated warning " Implicit conversion from float 11167.333333333332 to int loses precision"   
$sizes[0]=129 $sizes[1]=11437  (php getimagesize  reports the correct dimensions 16751x19001 )

When I comment out the resize stuff there is a black image created  (I Attached the image).


![17091284535385353799527929296507552950 (5)](https://github.com/ImageMagick/ImageMagick/assets/1076033/f6eb954a-bd72-45f2-b9aa-1f9ce45d09fc)

**Comment by snibgo** [comment]
I suppose that warning is from IMagick, not ImageMagick. IMagick is not supported by the same people as ImageMagick. I don't use IMagick. Perhaps you should ask the question elsewhere.

As it is a warning, perhaps it can be ignored.

I suggest you try an ImageMagick command:
```
magick in.tif -resize 86x7624 out.jpg
```
Does that work? If not, what error is raised? If you remove the resize, does it work?

As your input image is large, perhaps you are out of memory.

**Reply by eduh** [reply]
Since the whole conversion stuff is embedded in PHP I try to look for a solution without doing commandline stuff althought it is worth a try.  The black jpeg is a result for the script without the resize. The original tif is also converted to jp2000 with another tooklkit and can be viewed here:  https://objects.library.uu.nl/reader/viewer.php?obj=1874-433118&pagenum=21&lan=en
Memory is no problem. Runs in a large kubernetes environment and no problem there.
We use in PHP this version
imagick module => enabled
imagick module version => 3.7.0
imagick classes => Imagick, ImagickDraw, ImagickPixel, ImagickPixelIterator, ImagickKernel
Imagick compiled with ImageMagick version => ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagi
ck.org
Imagick using ImageMagick library version => ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25 https://imagemagi
ck.org

**Reply by Matiz7** [reply]
I use your command. like this .. why magick not exist
[bin]# magick /data/tmp/syn/2729656/fm_67b0ff04192c498a83734eb724b94ad5.tif -resize 1042*1371 590790_111.jpg
bash: magick: command not found...
[bin]# convert
Version: ImageMagick 6.9.12-50 Q16 x86_64 17254 https://legacy.imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
...

use convert command which from ImageMagick.  It donot work
[bin]# convert -resize 1042 /data/tmp/syn/2729656/fm_67b0ff04192c498a83734eb724b94ad5.tif /data/tmp/syn/2729656/590790_111.jpg
convert: Incompatible type for "RichTIFFIPTC"; tag ignored. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/898.

I dont know whether the latest version ImageMagick work well with TIF file

**Comment by snibgo** [comment]
You have an old version  of IM, v6. That uses `convert`, not `magick`.

> warning/tiff.c/TIFFWarnings/898.

That is a warning, not an error. Did it create the output file?

I suggest doing the operation in the order: read the input, then resize it, then write the output.
```
convert /data/tmp/syn/2729656/fm_67b0ff04192c498a83734eb724b94ad5.tif -resize 1042 /data/tmp/syn/2729656/590790_111.jpg
```
If that works (perhaps with a warning), then ImageMagick does not have a problem.

**Reply by eduh** [reply]
I tried to use convert and yes, that works. So perhaps the problem is in the PHP extension. Is there a seperate community where I can ask?

**Reply by Matiz7** [reply]
convert in.tif -resize 1042 out.jpg
Actually output created 2 different size files. dont know what differents it is
-rw-r--r--. 1 root  root    1126013 Dec 14 09:12 590790_111-0.jpg
-rw-r--r--. 1 root  root    1097660 Dec 14 09:12 590790_111-1.jpg
fix my script and It works again

**Reply by fmw42** [reply]
The message was only a warning about a TIF tag not conforming.

The TIFF file contains two different images. You likely just want the first. If that is the case use

`convert-quiet in.tif[0] -resize 1042 out.jpg`

The -quiet suppresses the warning.
The [0] says to use only the first page (image) in the TIFF file
